Lab2/app/src/main/java/nm/lab2/DatabaseFragment.java

```diff
@@ -0,0 +1,131 @@
+package nm.lab2;
+
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.webkit.WebView;
+import android.widget.Button;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+/**
+ * Fragment for image feed
+ */
+
+public class DatabaseFragment extends Fragment {
+    int current_image = 0;
+    int num_images = 0;
+    WebView webView;
+    ArrayList<String> images = new ArrayList<>();
+    FeedReaderDbHelper mDbHelper;
+
+    public DatabaseFragment() {
+    }
+
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_database, container, false);
+
+        // define parts of interface
+        Button searchFragmentButton = (Button) rootView.findViewById(R.id.search_fragment_button);
+        searchFragmentButton.setOnClickListener(new ChangeBackFragmentListener());
+        Button next = (Button) rootView.findViewById(R.id.next_button_database);
+        next.setOnClickListener(new NextListener());
+        Button previous = (Button) rootView.findViewById(R.id.previous_button_database);
+        previous.setOnClickListener(new PreviousListener());
+        Button delete = (Button) rootView.findViewById(R.id.delete_button);
+        delete.setOnClickListener(new DeleteListener());
+        webView = (WebView) rootView.findViewById(R.id.database_web_view);
+
+        //retrieve from database
+        getSavedImages();
+        //put first image in webView if it exists
+        if (num_images != 0){
+            webView.loadUrl(images.get(0));
+        } else{
+            Toast.makeText(getActivity().getBaseContext(), "No images have been saved",
+                    Toast.LENGTH_LONG).show();
+        }
+
+        return rootView;
+    }
+
+    //get images from database and save them into arraylist "images"
+    public void getSavedImages() {
+        mDbHelper = new FeedReaderDbHelper(getActivity()); //initialize database
```

Comment:
Its better to put this line in the onCreateView method since you know it is always called first. 
While in your case you know that this method is always called first you may keep working and forget that this method also initializes mDbHelper and try to use mDbHelper before this method is called and its initialized. 
Just a good practice to always initialize class fields in either the constructor or in this case the onCreateView